tapiriik/services/Setio/setio.py

```diff
@@ -0,0 +1,286 @@
+            headers = {
+                'content-type': "application/json",
+                'cache-control': "no-cache",
+            }
+            streamdata = requests.request("POST", url, data=payload, headers=headers)
+            if streamdata.status_code == 500:
+                raise APIException("Internal server error")
+
+            if streamdata.status_code == 403:
+                raise APIException("No authorization to download activity", block=True,
+                                   user_exception=UserException(UserExceptionType.Authorization,
+                                                                intervention_required=True))
+
+            if streamdata.status_code != 204: # "Record Not Found":
+                try:
+                    commentdata = streamdata.json()
+                except:
+                    raise APIException("Stream data returned is not JSON")
+
+                if "comment" in commentdata:
+                    activity.Notes = commentdata["comment"]
+                else:
+                    activity.Notes = ""
+            else:
+                activity.Notes = ""
+
+            activity.GPS = True
+
+            activity.Private = False
+            activity.Stationary = False #True = no sensor data
+
+            activity.CalculateUID()
+            activities.append(activity)
+
+        return activities, exclusions
+
+
+    def SubscribeToPartialSyncTrigger(self, serviceRecord):
+        # There is no per-user webhook subscription with Setio.
+        serviceRecord.SetPartialSyncTriggerSubscriptionState(True)
+
+    def UnsubscribeFromPartialSyncTrigger(self, serviceRecord):
+    #     # As above.
+        serviceRecord.SetPartialSyncTriggerSubscriptionState(False)
+
+    def DownloadActivity(self, svcRecord, activity):
+        # if activity.ServiceData["Manual"]:
+        #    # Maybe implement later
+        #    activity.Laps = [Lap(startTime=activity.StartTime, endTime=activity.EndTime, stats=activity.Stats)]
+        #    return activity
+        activityID = activity.ServiceData["ActivityID"]
+
+        extID = svcRecord.ExternalID
+
+        url = "https://us-central1-project-2489250248063150762.cloudfunctions.net/getRunData"
+        payload = "{\"userId\": \"" + extID + "\",\"runId\":\"" + str(activityID) + "\"}"
+        headers = {
+            'content-type': "application/json",
+            'cache-control': "no-cache",
+        }
+        streamdata = requests.request("POST", url, data=payload, headers=headers)
+        if streamdata.status_code == 500:
+            raise APIException("Internal server error")
+
+        if streamdata.status_code == 403:
+            raise APIException("No authorization to download activity", block=True, user_exception=UserException(UserExceptionType.Authorization, intervention_required=True))
+
+        if streamdata.status_code == 204: #"Record Not Found":

```

Comment:
Similarly, better to check that it's 200 rather than assume that it's one of the 4 you expect it to be.